byway/src/main/webapp/routepage.js

```diff
@@ -12,117 +12,198 @@
+ *  TODO: Delete markers for recommended stops not selected.
+ *  TODO: Disable usage after? Don't want to keep adding to list. 
+ */
+function generateRoute() {
+  route.splice(0,0,start);
+  route.splice(route.length, 0, end);
+  console.log(route);
+  renderRouteList();
+}
+
+/**
+ * Reorders the elements in route list based on the optimized order of 
+ * waypoints returned in the response 
+ * @param {response} response response from the directions service object
+ */
+function orderWaypoints(response){
+  let waypoint_order = response.routes[0].waypoint_order;
+  let route_copy = [...route];
+  for(let i = 0; i < route.length; i++){
+    route[i] = route_copy[waypoint_order[i]];
   }
 }
 
-/** Get the new list of stops from datastore onload */
-function getStopsOnload(){
-  clearStops();
+/**
+ * Calculates and sums up the distance and time duration between all destinations (legs)
+ * @param {response} response response from the directions service object
+ * @return {number, number, number} distance total driving distance for whole route, 
+ * hours estimated driving time in hours, minutes estimated driving time in minutes
+ */
+function computeDistanceTime(response) {
+  let totalDist = 0;
+  let totalTime = 0;
+  // full route
+  let route_response = response.routes[0];
+
+  for (let i = 0; i < route_response.legs.length; i++) {
+    // in meters
+    totalDist += route_response.legs[i].distance.value;
+    // in seconds
+    totalTime += route_response.legs[i].duration.value;
+  }
+  
+  let distance = (totalDist / 1000).toFixed(2);
+  let hours = Math.floor(totalTime / 3600);
+  let minutes = Math.round((totalTime - hours*3600) / 60);
+
+  return {distance, hours, minutes};
+}
+
+/**
+ * Updates total distance and duration elements in HTML
+ * @param {response} response response from the directions service object
+ */
+function updateDistanceTime(response){
+  let result = computeDistanceTime(response);
+  document.getElementById("distance").innerText = result.distance + "km";
+  document.getElementById("duration").innerText = 
+    (result.hours == 0) ? result.minutes + " mins" : result.hours + " hr " + result.minutes + " mins";
+}
+ 
+/** Clear the route panel in the html */
+function clearRoute(){
+  const routeList = document.getElementById('route-list');
+  if(routeList != null){
+    routeList.innerText = "";
+  }
+}
+ 
+/** Get trip info from datastore onload */
+function getRouteOnload(){
+  clearRoute();
   fetch('/api/stop')
   .then(response => response.json())
-  .then((stopsResponse) => {
-    if(stopsResponse != null){
-      stops.push(...stopsResponse);
+  .then((trip) => {
+    if(trip != null){
+      start = end = trip.start;
+      destinations = trip.destinations;
+      route.push(...trip.route)
+
+      calcRoute();
+    }
+    else{
+      console.log("Could not retrieve any routes nor destinations associated with this trip. Please reload page and try again.");
```

Comment:
Nit: split this into multiple lines so that it is less than 100 characters long

Reply:
Changed.